jdaviz/app.py

```diff
@@ -1849,6 +1852,96 @@ def _get_viewer_item(self, ref_or_id):
             viewer_item = self._viewer_item_by_id(ref_or_id)
         return viewer_item
 
+    def _reparent_subsets(self, old_parent, new_parent=None):
+        '''
+        Re-parent subsets that belong to the specified data
+
+        Parameters
+        ----------
+        old_parent : glue.core.Data, str
+            The item from the data collection off of which to move the subset definitions.
+
+        new_parent : glue.core.Data, str
+            The item from the data collection to make the new parent. If None, the first
+            item in the data collection that doesn't match ``old_parent`` will be chosen.
+        '''
+        from astropy.wcs.utils import pixel_to_pixel
+
+        if isinstance(old_parent, str):
+            old_parent = self.data_collection(old_parent)
+
+        if isinstance(new_parent, str):
+            new_parent = self.data_collection(new_parent)
+        elif new_parent is None:
+            for data in self.data_collection:
+                if data is not old_parent:
+                    new_parent = data
+                    break
+
+        # Set subset attributes to match a remaining data collection member.
+        for subset_group in self.data_collection.subset_groups:
+            # Flag to skip this
+            if subset_group.subset_state.attributes[0].parent is old_parent:
+                for att in ("att", "xatt", "yatt", "x_att", "y_att"):
+                    if hasattr(subset_group.subset_state, att):
+                        subset_att = getattr(subset_group.subset_state, att)
+                        data_components = new_parent.components
+                        if subset_att not in data_components:
+                            cid = [c for c in data_components if c.label == subset_att.label][0]
+                            setattr(subset_group.subset_state, att, cid)
+
+                # Translate bounds through WCS if needed
+                if (self.config == "imviz" and
+                        self._jdaviz_helper.plugins["Links Control"].link_type == "WCS"):
+                    # Get the correct link to use for translation
+                    roi = subset_group.subset_state.roi
+                    if type(roi) in (CircularROI, CircularAnnulusROI,
+                                     EllipticalROI, TrueCircularROI):
+                        old_xc = roi.xc
+                        old_yc = roi.yc
+                        # Convert center
+                        x, y = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                              roi.xc, roi.yc)
+                        # Can't use set_center here because CircularAnnulusROI doesn't have it
+                        roi.xc = x
+                        roi.yc = y
```

Comment:
@rosteen , do you not want to simplify this part for recentering?

Reply:
Yes, this was the "almost" in "addressed almost all your comments" 🙂. My attention is getting pulled in too many directions today...I'm going to keep the if statement since as I said below, we do need the rest of the logic for updating things other than the center.